simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -1,50 +1,133 @@
-
+from ...utils.code_utils import (
+    deprecate_class,
+    validate_string,
+    validate_ndarray_with_shape,
+)
+import warnings
 import numpy as np
 from scipy.constants import mu_0
-
 from ...survey import BaseRx
 
 
 def _alpha(src):
     return 1 / (2 * np.pi * mu_0 * src.frequency)
 
 
-class PointNaturalSource(BaseRx):
-    """Point receiver class for magnetotelluric simulations.
+# THIS SHOULD BE HIDDEN METHOD
+def _getP(rx, mesh, projected_grid, field="e", is_tipper_bs=False):
+    """Projection matrix from discrete field solution to measurement locations.
+
+    Note projection matrices are stored as a dictionary listed by meshes.
+
+    Parameters
+    ----------
+    rx : .natural_source.receivers.Impedance
+        a NSEM receiver
+    mesh : discretize.base.BaseMesh
+        The mesh on which the discrete set of equations is solved.
+    projected_grid : str
+        Define what part of the mesh (i.e. edges, faces, centers, nodes) to
+        project from. Must be one of::
+
+            'Ex', 'edges_x'           -> x-component of field defined on x edges
+            'Ey', 'edges_y'           -> y-component of field defined on y edges
+            'Ez', 'edges_z'           -> z-component of field defined on z edges
+            'Fx', 'faces_x'           -> x-component of field defined on x faces
+            'Fy', 'faces_y'           -> y-component of field defined on y faces
+            'Fz', 'faces_z'           -> z-component of field defined on z faces
+            'N', 'nodes'              -> scalar field defined on nodes
+            'CC', 'cell_centers'      -> scalar field defined on cell centers
+            'CCVx', 'cell_centers_x'  -> x-component of vector field defined on cell centers
+            'CCVy', 'cell_centers_y'  -> y-component of vector field defined on cell centers
+            'CCVz', 'cell_centers_z'  -> z-component of vector field defined on cell centers
+
+    field : {"e", "h"}
+        Whether to project electric or magnetic fields from the mesh.
+    is_tipper_bs : bool
+        Whether the projection is for a remote base station location where horizontal
+        magnetic fields are measured. Ensures stash names for projection matrices
+        aren't reused.
+
+    Returns
+    -------
+    sp.sparse.csr_matrix
+        The projection matrix.
+    """
+    if mesh.dim < 3:
+        return rx.getP(mesh, projected_grid)
+
+    if is_tipper_bs:
+        stash_name = projected_grid + "_bs"
+    else:
+        stash_name = projected_grid
+
+    if (mesh, stash_name) in rx._Ps:
+        return rx._Ps[(mesh, stash_name, field)]
+
+    if field == "e":
+        locs = rx.locations_e
+    elif field == "h":
+        if isinstance(rx, ApparentConductivity):
+            locs = rx.locations_h
+        elif is_tipper_bs:
+            locs = rx.locations_bs
+        else:
+            locs = rx.locations
+    else:
+        raise ValueError("Field type {} unrecognized. Use 'e' or 'h'".format(field))
```

Comment:
Small change to use f-string here.
```suggestion
        raise ValueError(f"Field type {field} unrecognized. Use 'e' or 'h'")
```